integration/tests/params.go

```diff
@@ -0,0 +1,163 @@
+// +build integration
+
+package tests
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/weaveworks/eksctl/integration/runner"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/utils/names"
+)
+
+// Params groups all test parameters.
+type Params struct {
+	EksctlPath string
+	Region     string
+	Version    string
+	// Flags to help with the development of the integration tests
+	clusterNamePrefix string
+	ClusterName       string
+	DoCreate          bool
+	DoDelete          bool
+	KubeconfigPath    string
+	KubeconfigTemp    bool
+	TestDirectory     string
+	// privateSSHKeyPath is the SSH key to use for Git operations.
+	PrivateSSHKeyPath       string
+	EksctlCmd               runner.Cmd
+	EksctlCreateCmd         runner.Cmd
+	EksctlGetCmd            runner.Cmd
+	EksctlDeleteCmd         runner.Cmd
+	EksctlDeleteClusterCmd  runner.Cmd
+	EksctlScaleNodeGroupCmd runner.Cmd
+	EksctlUtilsCmd          runner.Cmd
+	EksctlExperimentalCmd   runner.Cmd
+	// Keep track of created clusters, for post-tests clean-up.
+	clustersToDelete []string
+}
+
+// SetRegion sets the provided region and re-generates other fields depending
+// on the region field.
+func (p *Params) SetRegion(region string) {
+	p.Region = region
+	p.GenerateCommands()
+}
+
+// GenerateKubeconfigPath generates a path in ${TEMP} based on these params' cluster name.
+func (p *Params) GenerateKubeconfigPath() {
+	p.KubeconfigPath = path.Join(os.TempDir(), fmt.Sprintf("%s.yaml", p.ClusterName))
+}
+
+// GenerateCommands generates eksctl commands with the various options & values
+// provided to this `Params` object.
+func (p *Params) GenerateCommands() {
+	p.EksctlCmd = runner.NewCmd(p.EksctlPath).
```

Comment:
Just out of curiosity, are the Command objects something that should be reused for some reason? Why create them once and reuse them instead of always returning a new one with something like:

```
func (p *Params) newCreateCommand() (p *runner.Command)  {
  return runner.NewCmd(p.EksctlPath).
      WithArgs("create").
      WithTimeout(25 * time.Minute)
}
```

Reply:
I am actually not sure as this precedes me, and I didn't look too deep into what they do.